api/src/Serializer/Normalizer/RelatedCollectionLinkNormalizer.php

```diff
@@ -170,21 +178,35 @@ public function getRelatedCollectionHref($object, $rel, array $context = []): st
         $relatedFilterName ??= $relationMetadata['inversedBy'];
 
         if (empty($relatedResourceClass) || empty($relatedFilterName)) {
-            throw new UnsupportedRelationException('The '.$resourceClass.'#'.$rel.' relation does not have both a targetEntity and a mappedBy or inversedBy property');
+            return [false, 'The '.$resourceClass.'#'.$rel.' relation does not have both a targetEntity and a mappedBy or inversedBy property'];
         }
 
-        $resourceMetadataCollection = $this->resourceMetadataCollectionFactory->create($relatedResourceClass);
-        $operation = OperationHelper::findOneByType($resourceMetadataCollection, GetCollection::class);
+        $lookupKey = $relatedResourceClass.':'.$relatedFilterName;
+        if (isset($this->exactSearchFilterExistsCache[$lookupKey])) {
+            $result = $this->exactSearchFilterExistsCache[$lookupKey];
+        } else {
+            $result = [null, ''];
```

Comment:
a union type would be enough.
sadly, both types are string...but never both keys are set.
Maybe 2 wrapper classes around strings would be more explicit.

Reply:
no, they are not both strings; I have converted it to a union type. thx